File: zerojudge/a006-一元二次方程式/solution.cpp
/*
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem link: https://zerojudge.tw/ShowProblem?problemid=a006
 * Status: AC
 */

#include <cmath>
#include <iostream>

using namespace std;

void discriminant(int, int, int);

int main() {
    int a, b, c;
    cin >> a >> b >> c;

    discriminant(a, b, c);
    return 0;
}

void discriminant(int a, int b, int c) {
    int d = (b * b - 4 * a * c);

    if (d > 0) {
        int x1 = ((-b + sqrt(d)) / 2 * a);
        int x2 = ((-b - sqrt(d)) / 2 * a);
        cout << "Two different roots x1=" << x1 << " , " << "x2=" << x2 << endl;
    } else if (d == 0) {
        int x = ((-b) / (2 * a));
        cout << "Two same roots x=" << x << endl;
    } else {
        cout << "No real root" << endl;
    }

    return;
}

File: zerojudge/a006-一元二次方程式/solution.py
"""
* Author: 1chooo<hugo970217@gmail.com>
* Problem link: https://zerojudge.tw/ShowProblem?problemid=a006
* Status: AC
"""

inputString = str(input()).split(" ")

a = int(inputString[0])
b = int(inputString[1])
c = int(inputString[2])


def discriminant(a, b, c):
    d = int(b * b - 4 * a * c)  # It can also be written as integer.

    if d > 0:
        x1 = int((-b + d ** (1 / 2)) / 2 * a)
        x2 = int((-b - d ** (1 / 2)) / 2 * a)
        print(f"Two different roots x1={x1} , x2={x2}")
    elif d == 0:
        x = int((-b) / (2 * a))
        print(f"Two same roots x={x}")
    else:
        print("No real root")
    return


if __name__ == "__main__":
    discriminant(a, b, c)
